lib/syncthing/utils.go

```diff
@@ -124,10 +123,6 @@ func copyFile(src, dst string) error {
 	return nil
 }
 
-func OpenGoleveldb(path string, tuning config.Tuning) (*db.Lowlevel, error) {
-	ldb, err := backend.Open(path, backend.Tuning(tuning))
-	if err != nil {
-		return nil, err
-	}
-	return db.NewLowlevel(ldb), nil
+func OpenDBBackend(path string, tuning config.Tuning) (backend.Backend, error) {
```

Comment:
This function doesn't add much value. We could do the cast in main...

Reply:
The intention is that all parameters of `syncthing.New` can be created from this package alone, i.e. no need to import `lib/db/backend`. And I am not sure anymore, but I think it might even be a bit more than just convenience: I believe creating reverse bindings for a package puts some restrictions on its exported functions, but not on dependent packages. And afaik I made sure lib/syncthing conforms to those, but not any others.

Reply:
Fair 'nuff